drop-transfer/src/ws/server/v4.rs

```diff
@@ -444,16 +550,21 @@ impl handler::Downloader for Downloader {
                     meta.len()
                 );
 
-                let report = self.request_csum(meta.len()).await?;
-
-                if report.limit == meta.len() && report.checksum == csum {
-                    // All matches, we can continue with temp file
+                if meta.len() < task.file.size() {
+                    let report = self.request_csum(meta.len()).await?;
+
+                    if report.limit == meta.len() && report.checksum == csum {
+                        // All matches, we can continue with temp file
+                        self.offset = meta.len();
+                    } else {
+                        info!(
+                            self.logger,
+                            "Found missmatch in partially downloaded file, overwriting"
+                        );
+                    }
+                } else if meta.len() == task.file.size() && self.full_csum.get().await == csum {
```

Comment:
The `else` clause is missing, I assume that it just leaves `self.offset` at default(0) and it just works?

Do we have write_at_offset or do we just append the bytes at the end? In such case, if the temp file would be bigger somehow we would just append the whole file to what's already there?

Reply:
`else` is not needed, it will default to offset = 0.
We always append at the end of the temp file, but we ensure the size and integrity first so that we're sure it exactly matches the target file

Reply:
Explicitness is always nice especially if for some weird reason, we would initialize `offset` to a non-zero value in the future

Reply:
Did just that. Now it's set in all the branches